reconcile/utils/terrascript_client.py

```diff
@@ -803,6 +803,8 @@ def populate_tf_resources(self, populate_spec, existing_secrets,
             self.populate_tf_resource_service_account(resource,
                                                       namespace_info,
                                                       ocm_map=ocm_map)
+        elif provider == 'aws-iam-role':
+            self.populate_tf_resource_role(resource, namespace_info)
```

Comment:
I guess we could turn this into a dictionary lookup and a dispatcher, as part of https://issues.redhat.com/projects/APPSRE/issues/APPSRE-3505 .

Reply:
for consistency, this class has no tests until now. not easy to test. happy to do this separately though.